### ISSUE TITLE
Cant use multithread with brain.recurrent.ltsm

### ISSUE THREAD

**Issue by ArtichautDev** [BrainJS/brain.js#786]
![](https://c.tenor.com/H9Yz02w9AvEAAAAC/baby-yoda-cat.gif)

## _What_ is wrong?

training network takes really long and I need to use multi thread --> 
this don't work for me : https://github.com/BrainJS/brain.js/issues/417

## _Where_ does it happen?

In the a NeuralNetwork when trying to run a net in node.js on my mac

## _How_ do we replicate the issue?

use my code: 

const brain = require('brain.js');
const net = new brain.recurrent.LSTM();
console.log('starting IA train machine learning');
net.train([
    { input: tokengrab1title, output: 'tokengrab' },
    { input: nottokengrab1title, output: 'nottokengrab' },
    { input: nottokengrab2title, output: 'nottokengrab' },
    { input: nottokengrab3title, output: 'nottokengrab' },
    { input: nottokengrab4title, output: 'nottokengrab' },
    { input: nottokengrab5title, output: 'nottokengrab' },
    { input: nottokengrab6title, output: 'nottokengrab' },
    { input: nottokengrab7title, output: 'nottokengrab' },
    { input: nottokengrab8title, output: 'nottokengrab' },
    { input: nottokengrab9title, output: 'nottokengrab' },
    { input: nottokengrab10title, output: 'nottokengrab' },
  
]);

- when I try to set this in multi thread its don't work

### Nodejs:
16

### Brain.js:
latest
## _How_ important is this (1-5)?
4

## Other Comments
someone can do this for me because I have spent 5h on this...

Artichaut#5655 for rapid contact

**Comment by robertleeplummerjr** [comment]
>when I try to set this in multi thread its don't work

What do you mean, multi-threaded?

**Comment by ArtichautDev** [comment]
I want to use multiple core for my code with brain.js

https://cdn.discordapp.com/attachments/763849429320990751/964196204504305785/Screen_Shot_2022-04-14_at_6.11.06_PM.png

its using only 100% cpu I want to use all my cores (1500%) (100%= 1 core)

and its take a very long time to train brain

**Comment by robertleeplummerjr** [comment]
There isn't currently a feature that currently utilize multi-threading.  The exceptions are somewhat `NeuralNetworkGPU`, `FeedForward`, and, `Recurrent` that can utilize the GPU.  There was some speculation as to how that feature could work but the problem you'll run into when you go to implement is backpropagation.

Outside of Brain.js: You can get around this by utilizing different types of backpropagation and splitting up your training set to a smaller subset of training data for each thread, but you run into the issue of later how do you combine your weights for reuse?  This is where most of the experimentation leaves off.  I've not seen a really good resilient solution just yet that would take advantage of multi threading in node for this task.
